[K9VULN-16251] Add agentic Bits AI docs (#37511)
DataDog/documentation: content/en/security/code_security/static_analysis/ai_enhanced_sast.md

Each finding includes a section with an explanation of the assessment. You can provide Bits AI with feedback on its assessment using a thumbs up &#128077; or thumbs down &#128078;.
{{< img src="/code_security/static_analysis/false_positive_filtering_sast_side_panel_higher_res_png.png" alt="Visual indicator of a false positive assessment in SAST side panel" style="width:100%;">}}

### Agentic Bits AI assessments

> For supported SAST rules, Bits AI Assessments use an agentic approach to gather repository context before classifying findings. Bits AI can read related files and search for symbols and patterns. It can also inspect nearby directory structure to verify definitions, call paths, sanitizers, and framework wiring that are not visible in a single file.

Reviewer: ```suggestion
For supported SAST rules, Bits AI Assessments use an agentic approach to gather repository context before classifying findings. Bits AI can read related files and search for symbols and patterns. Bits AI can also inspect the surrounding directory structure to verify definitions, call paths, sanitizers, and framework wiring that are not visible in a single file.
```

> For supported SAST rules, Bits AI Assessments use an agentic approach to gather repository context before classifying findings. Bits AI can read related files and search for symbols and patterns. It can also inspect nearby directory structure to verify definitions, call paths, sanitizers, and framework wiring that are not visible in a single file.

Reviewer: Capitalization of "assessments" is inconsistent. The heading (line 146) uses lowercase "assessments" (correct sentence case), but the body capitalizes it as "Assessments." If "Bits AI Assessments" is an official feature name, it should be capitalized throughout.

> This extra context helps Bits AI distinguish true positives from false positives for findings that depend on cross-file behavior. Agentic Bits AI Assessments apply to SAST findings only.

Reviewer: ```suggestion
The additional repository context helps Bits AI distinguish true positives from false positives for findings that depend on cross-file behavior. Agentic Bits AI Assessments apply to SAST findings only.
```

In the SAST rule side panel, expand the false positive reports accordion to review reports shared by your organization for the selected rule. Use the custom context tab in the same section to add guidance for future Bits AI assessments. Memories apply at the organization and rule level for SAST. They apply only to security category SAST rules in Datadog's default rulesets and do not apply to custom rules.

### Agentic false positive filtering

For supported SAST rules, Bits AI uses an agentic approach to gather repository context before classifying findings. Bits AI can read related files and search for symbols and patterns. It can also inspect nearby directory structure to verify definitions, call paths, sanitizers, and framework wiring that are not visible in a single file.

> This extra context helps Bits AI distinguish true positives from false positives for findings that depend on cross-file behavior. Agentic false positive filtering applies to SAST findings only.

Reviewer: ```suggestion
The additional repository context helps Bits AI distinguish true positives from false positives for findings that depend on cross-file behavior. Agentic false positive filtering applies to SAST findings only.
```

> This extra context helps Bits AI distinguish true positives from false positives for findings that depend on cross-file behavior. Agentic false positive filtering applies to SAST findings only.

Reviewer: ```suggestion
This extra context helps Bits AI distinguish true positives from false positives for findings that depend on cross-file behavior. Agentic false positive filtering applies only to SAST findings.
```

> This extra context helps Bits AI distinguish true positives from false positives for findings that depend on cross-file behavior. Agentic false positive filtering applies only to SAST findings.

Reviewer: ```suggestion
For supported SAST rules, Bits AI uses an agentic approach to gather repository context before classifying findings. Bits AI can read related files and search for symbols and patterns. It can also inspect the surrounding directory structure to verify definitions, call paths, sanitizers, and framework wiring that are not visible in a single file.

The additional repository context helps Bits AI distinguish true positives from false positives for findings that depend on cross-file behavior. Agentic false positive filtering applies only to SAST findings.
```
